library/src/scripts/forms/themeEditor/ColorPicker.tsx

```diff
@@ -0,0 +1,154 @@
+                    Validator.isRgba(color) ||
+                    Validator.isHsl(color) ||
+                    Validator.isHsla(color))
+            );
+        }
+    };
+
+    const clickReadInput = () => {
+        if (colorInput && colorInput.current) {
+            colorInput.current.click();
+        }
+    };
+
+    const onTextInputChange = e => {
+        const colorString = e.target.value;
+        if (isValidColorString(colorString)) {
+            const newColor = color(colorString);
+            selectedColorHelpers.setValue(newColor);
+            setLastValidColor(newColor);
+        }
+        setInputTextValue(e.target.value);
+    };
+
+    const onColorInputChange = e => {
+        const newColor = color(e.target.value);
+        selectedColorHelpers.setValue(newColor);
+        setLastValidColor(newColor);
+        setInputTextValue(newColor.toString());
+    };
+
+    const formattedColor = colorOut(lastValidColor);
+    const hasError = props.errors && props.errors.length > 0;
+
+    return (
+        <>
+            <span className={classes.root}>
+                {/*"Real" color input*/}
+                <input
+                    {...props.inputProps}
+                    ref={colorInput}
+                    type="color"
+                    id={props.inputID}
+                    aria-describedby={props.labelID}
+                    className={classNames(classes.realInput, visibility().visuallyHidden)}
+                    onChange={onColorInputChange}
+                    aria-errormessage={hasError ? errorID : undefined}
+                    defaultValue={initialColor.toString()}
+                />
+                {/*Text Input*/}
+                <input
+                    ref={textInput}
+                    type="text"
+                    aria-describedby={props.labelID}
+                    aria-hidden={true}
+                    className={classNames(classes.textInput, {
+                        [classes.invalidColor]: !isValidColorString(inputTextValue),
+                    })}
+                    placeholder={"#0291DB"}
+                    value={inputTextValue}
+                    onChange={onTextInputChange}
+                />
+                {/*Swatch*/}
+                <Button
+                    onClick={clickReadInput}
+                    style={{ backgroundColor: isValidColorString(formattedColor) ? formattedColor : "#000" }}
+                    title={formattedColor}
+                    aria-hidden={true}
+                    className={classes.swatch}
+                    tabIndex={-1}
+                    baseClass={ButtonTypes.CUSTOM}
+                >
+                    <span className={visibility().visuallyHidden}>{formattedColor}</span>
+                </Button>
+            </span>
+            {props.errors && props.errors.length > 0 && (
+                <ul id={errorID} className={builderClasses.errorContainer}>
+                    {props.errors.map((error, i) => {
```

Comment:
About the error messages here:

- What's the scenario that we would have multiple errors on a single input? The API certainly won't be giving any back. The only that I could see happening are invalid colors which could be validated here.
- It would probably be better to store the errors in formik so they can also be used for top level form validation. You use it inside of an `onBlur` handler and also run it when the component first mounts. Something like this:

```
const validateColor = () => {
	//... Some conditional
	helper.setFieldError("Invalid Color");
}

useEffect(() => {
    validateColor()
    // eslint-disable-next-line react-hooks/exhaustive-deps
}, []);

<input

   // ... other props
	onBlur={(e) => {
		selectedColor.onBlur(e);
		validateColor();
    }}
>

```